Shopist/Shopist/Scenes/ListViewController.swift

```diff
@@ -0,0 +1,82 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import UIKit
+
+class ListViewController: UICollectionViewController {
+    private static let cellIdentifier = "cell"
+    private let api = API()
+
+    convenience init() {
+        self.init(collectionViewLayout: UICollectionViewFlowLayout())
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        let refreshControl = UIRefreshControl()
+        refreshControl.addTarget(self, action: #selector(refresh), for: .valueChanged)
+        collectionView.refreshControl = refreshControl
+        collectionView.backgroundColor = .white
+        collectionView.register(Cell.self, forCellWithReuseIdentifier: Self.cellIdentifier)
+        collectionView.delegate = self
+        setupLayout(for: view.bounds.size)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        refresh(force: true)
+    }
+
+    func fetch(with api: API) {
+        // to be overridden
+        return
+    }
+
+    func config(cell: Cell, for index: Int) {
+        // to be overridden
+        return
+    }
+
+    @objc
+    private func refresh(force: Bool = false) {
+        let refreshNeeded = force || (self.collectionView.refreshControl?.isRefreshing ?? true)
+        if !refreshNeeded { return }
+        fetch(with: api)
+    }
+
+    func reload() {
+        DispatchQueue.main.async {
+            self.collectionView.refreshControl?.endRefreshing()
+            self.collectionView.reloadData()
+        }
+    }
```

Comment:
Is pull to refresh feature necessary for demo flow? It brings complexity, so if it's unnecessary let's avoid adding it. I don't see it on Shopist website, but dunno if we leverage it on Android.